File: invision-backend/app/Enums/NotificationType.php
<?php

namespace App\Enums;

enum NotificationType: string
{
    case Task = 'task';
    case Message = 'message';
    case Alert = 'alert';
    case Announcement = 'announcement';
    case System = 'system';

    public function label(): string
    {
        return match ($this) {
            self::Task => 'Task',
            self::Message => 'Message',
            self::Alert => 'Alert',
            self::Announcement => 'Announcement',
            self::System => 'System',
        };
    }

    public function color(): string
    {
        return match ($this) {
            self::Task => 'blue',
            self::Message => 'green',
            self::Alert => 'red',
            self::Announcement => 'purple',
            self::System => 'gray',
        };
    }

    public function icon(): string
    {
        return match ($this) {
            self::Task => 'clipboard-check',
            self::Message => 'chat-bubble-left',
            self::Alert => 'exclamation-triangle',
            self::Announcement => 'megaphone',
            self::System => 'cog',
        };
    }
}

File: invision-mobile/lib/src/core/enums/notification_type.dart
import 'package:flutter/material.dart';

enum NotificationType {
  task,
  message,
  alert,
  announcement,
  system;

  static NotificationType fromString(String value) {
    return switch (value) {
      'task' => NotificationType.task,
      'message' => NotificationType.message,
      'alert' => NotificationType.alert,
      'announcement' => NotificationType.announcement,
      'system' => NotificationType.system,
      _ => NotificationType.system,
    };
  }

  String get value => switch (this) {
    NotificationType.task => 'task',
    NotificationType.message => 'message',
    NotificationType.alert => 'alert',
    NotificationType.announcement => 'announcement',
    NotificationType.system => 'system',
  };

  String get label => switch (this) {
    NotificationType.task => 'Task',
    NotificationType.message => 'Message',
    NotificationType.alert => 'Alert',
    NotificationType.announcement => 'Announcement',
    NotificationType.system => 'System',
  };

  Color get color => switch (this) {
    NotificationType.task => Colors.blue,
    NotificationType.message => Colors.green,
    NotificationType.alert => Colors.red,
    NotificationType.announcement => Colors.purple,
    NotificationType.system => Colors.grey,
  };

  IconData get icon => switch (this) {
    NotificationType.task => Icons.task_alt,
    NotificationType.message => Icons.message,
    NotificationType.alert => Icons.warning,
    NotificationType.announcement => Icons.campaign,
    NotificationType.system => Icons.settings,
  };
}
